### ISSUE TITLE
[promethus-snmp-exporter] Unable to use configmap to load/reload configuration

### ISSUE THREAD

**Issue by tuctboh** [prometheus-community/helm-charts#3621]
### Describe the bug a clear and concise description of what the bug is.

I installed the chart using -

helm install prometheus-snmp-exporter --set config=" " prometheus-community/prometheus-snmp-exporter

I realize thats not proper, but I had tried everything else I could think of to get the configmap option to kick in. It kept failing so I did that and it worked. GREAT!

So then I started to try to edit the config map and then ran into problems. I first tried with the stock supplied snmp.yml, and no luck, I was told the file was too big. So I cut the file down to just ONE auth (V2) and just ONE module (if_mib). That worked, but then it never showed up as usable in Prometheus. I took a look at the log, and it ends with -

ts=2023-07-20T16:30:41.186Z caller=main.go:186 level=error msg="Error reloading config" err="yaml: unmarshal errors:\n  line 3: field TBOH_v2 not found in type config.plain\n  line 10: field if_mib not found in type config.plain"

TBOH_v2 was my auth config, if_mib was my module config. 

I logged on with a shell and took a look at the snmp.yml its using, and it looks right, starts off

```
# WARNING: This file was auto-generated using snmp_exporter generator, manual changes will be lost.
auths:
  TBOH_v2:
    community: NunyaBizne$$
    security_level: noAuthNoPriv
    auth_protocol: MD5
    priv_protocol: DES
    version: 2
modules:
  if_mib:
    walk:
```
which to the best of my extremely limited knowledge is fine.

So no snmp for me right now

### What's your helm version?

version.BuildInfo{Version:"v3.12.0", GitCommit:"c9f554d75773799f72ceef38c51210f1842a1dea", GitTreeState:"clean", GoVersion:"go1.20.3"}

### What's your kubectl version?

WARNING: This version information is deprecated and will be replaced with the output from kubectl version --short.  Use --output=yaml|json to get the full version. Client Version: version.Info{Major:"1", Minor:"26", GitVersion:"v1.26.5+k3s1", GitCommit:"7cefebeaac7dbdd0bfec131ea7a43a45cb125354", GitTreeState:"clean", BuildDate:"2023-05-26T22:50:26Z", GoVersion:"go1.19.9", Compiler:"gc", Platform:"linux/arm64"} Kustomize Version: v4.5.7 Server Version: version.Info{Major:"1", Minor:"26", GitVersion:"v1.26.5+k3s1", GitCommit:"7cefebeaac7dbdd0bfec131ea7a43a45cb125354", GitTreeState:"clean", BuildDate:"2023-05-26T22:50:26Z", GoVersion:"go1.19.9", Compiler:"gc", Platform:"linux/arm64"}

### Which chart?

prometheus-snmp-exporter

### What's the chart version?

prometheus-snmp-exporter-1.6.0	v0.21.0

### What happened?

Just isn't loading the config despite it looking right.

### What you expected to happen?

Load the config and answer Prometheus requests

### How to reproduce it?

Should be able to follow from above.
Download stock snmp.yaml
Change community_v2 to TBOH_v2
Change community_v2 community from public to something else
Remove community_v1 block
Remove all ids in modules block except if_mib

### Enter the changed values of values.yaml?

config:" "

### Enter the command that you execute and failing/misfunctioning.

helm install prometheus-snmp-exporter --set config=" " prometheus-community/prometheus-snmp-exporter

### Anything else we need to know?

I really appreciate the work everyone puts into it.

**Comment by tuctboh** [comment]
So I figured out the issue. Even those this was recently updated, it only went to 0.21.0, not 0.23.0 where the new config file was implemented.

Can it be brought current? :)

Tnx

**Comment by walker-tom** [comment]
Closing this as it was resolved by #4043, if you have any further problems or feature requests please open a new issue, thanks!
